js/piwik.js

```diff
@@ -3223,7 +3225,14 @@ if (typeof window.Piwik !== 'object') {
                     (msToExpire ? ';expires=' + expiryDate.toGMTString() : '') +
                     ';path=' + (path || '/') +
                     (domain ? ';domain=' + domain : '') +
-                    (secure ? ';secure' : '');
+                    (isSecureCookie ? ';secure' : '');
+            }
+
+            /*
+             * Set cookie secure flag
+             */
+            function setSecureCookie(value) {
+                isSecureCookie = value == 1 ? true : false;
```

Comment:
JSLint test fails here. Is there any reason for comparing `value` with `1`? If you want to ensure `isSecureCookie` contains a Boolean afterwards, you could also cast `value` to Boolean or simply use `isSecureCookie = !!value`, which should pass JSLint.